pkgs/development/python-modules/solidpython/default.nix

```diff
@@ -0,0 +1,42 @@
+{ lib
+, buildPythonPackage
+, pythonRelaxDepsHook
+, fetchPypi
+, poetry-core
+, pypng
+, ply
+, wcwidth
+, prettytable
+, euclid3
+}:
+
+buildPythonPackage rec {
+  pname = "solidpython";
+  version = "1.1.3";
+  format = "pyproject";
+
+  src = fetchPypi {
+    inherit pname version;
+    hash = "sha256-GgH5msqAL9oKjyBX29mnvy7XsB4iX69UmPk7M3vEEsg=";
+  };
+
+  propagatedBuildInputs = [ prettytable euclid3 pypng ply wcwidth ];
+
+  nativeBuildInputs = [ poetry-core pythonRelaxDepsHook ];
+  pythonRelaxDeps = [ "PrettyTable" "pypng" ];
+
+  postPatch = ''
+    substituteInPlace pyproject.toml \
+      --replace "poetry.masonry.api" "poetry.core.masonry.api"
+  '';
+
+  pythonImportsCheck = [ "solid" ];
+
```

Comment:
are we running those tests https://github.com/SolidCode/SolidPython/tree/master/solid/test ?

Reply:
Good catch, I added `unittestCheckHook` (and a minor patch) in order to run those tests